tests/s3_data_plane_tests.c

```diff
@@ -889,6 +889,312 @@ static int s_test_s3_existing_host_entry(struct aws_allocator *allocator, void *
     return 0;
 }
 
+static int s_s3_next_request_cancel_in_the_middle(
+    struct aws_s3_meta_request *meta_request,
+    struct aws_s3_request **out_request) {
+    AWS_PRECONDITION(meta_request);
+    AWS_PRECONDITION(out_request);
+
+    struct aws_s3_request *request = NULL;
+    struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
+
+    int result = AWS_OP_SUCCESS;
+
+    /* TODO: haven't dived into it, but assuming the finish state has been handled well */
+    aws_s3_meta_request_lock_synced_data(meta_request);
```

Comment:
It's a little cumbersome, but you should be able to re-run all of the original logic by looking up the original vtable:

 struct aws_s3_meta_request_test_results *results = meta_request->user_data;
 struct aws_s3_tester *tester = results->tester;
 struct aws_s3_meta_request_vtable *original_meta_request_vtable =
        aws_s3_tester_get_meta_request_vtable_patch(tester, 0)->original_vtable;

(Where 0 is the index of the vtable that has been patched.  If you are only patching one vtable, this can just be 0.)

Reply:
If we need to do this in a next pass, can we at least update the copy pasta and add a TODO? :)

Reply:
I have removed this copy/paste.